### ISSUE TITLE
PHP 7 compatibility

### ISSUE THREAD

**Issue by Dyr-Fenrir** [paypal/ipn-code-samples#91]
### General information

* Environment:  Sandbox
* `PayPal-Debug-ID` values:  N/A
* Language, language version, and OS:  MySQL 5.7 and PHP 7

### Issue description

Greetings,

Is it possible that the PaypalIPN.php is not compatible with PHP 7?

I do not know if it is related, but I know that magic_quotes is no longer compatible and I saw it was used in the code.

I get an error 500 on my ipn_handler.php since it refers to PaypalIPN.php, if I remove the all the code from ipn_handler.php and only keep `<?php header("HTTP/1.1 200 OK");?>` the IPN goes through correctly.

Can you check this out please?

Sorry that I could not provide more information, I am not that familiar with PHP.

Thank you for your time and help.

**Comment by overint** [comment]
I've used/tested it on PHP7 multiple times and it's worked fine for me.
Could you turn on and check your PHP error logs? A 500 error could mean anything.

**Comment by Dyr-Fenrir** [comment]
Greetings,

I found the problem, it was 100% my fault, sorry about that (I did not even think about checking the logs...).

There was some issues with the PHP 7 cURL plugin that I needed to fix (Windows server).

I fixed these (installation problems) and now it's working correctly.

- I needed to add the path to my php folder in my Windows' system environment variable: PATH.
- I needed to reboot for that change to work.
- I needed to fix the syntax of `curl_setopt($ch, CURLOPT_CAINFO, __DIR__ . "assets/cert/cacert.pem");` to `curl_setopt($ch, CURLOPT_CAINFO, __DIR__ . "\\assets\\cert\\cacert.pem");`.

It's working correctly now.

Thank you again for your time and help, it's greatly appreciated!
